<launch>
  <arg name="input/rois_number" default="6"/>
  <arg name="input/rois0" default="/perception/object_recognition/detection/mask0"/>
  <arg name="input/camera_info0" default="/sensing/camera/traffic_light0/camera_info"/>
  <arg name="input/rois1" default="/perception/object_recognition/detection/mask1"/>
  <arg name="input/camera_info1" default="/sensing/camera/traffic_light1/camera_info"/>
  <arg name="input/rois2" default="/perception/object_recognition/detection/mask2"/>
  <arg name="input/camera_info2" default="/sensing/camera/traffic_light2/camera_info"/>
  <arg name="input/rois3" default="/perception/object_recognition/detection/mask3"/>
  <arg name="input/camera_info3" default="/sensing/camera/traffic_light3/camera_info"/>
  <arg name="input/rois4" default="/perception/object_recognition/detection/mask4"/>
  <arg name="input/camera_info4" default="/sensing/camera/traffic_light4/camera_info"/>
  <arg name="input/rois5" default="/perception/object_recognition/detection/mask5"/>
  <arg name="input/camera_info5" default="/sensing/camera/traffic_light5/camera_info"/>
  <arg name="input/rois6" default="/perception/object_recognition/detection/mask6"/>
  <arg name="input/camera_info6" default="/sensing/camera/traffic_light6/camera_info"/>
  <arg name="input/rois7" default="/perception/object_recognition/detection/mask7"/>
  <arg name="input/camera_info7" default="/sensing/camera/traffic_light7/camera_info"/>
  <arg name="input/clusters" default="/perception/object_recognition/detection/clustering/clusters"/>
  <arg name="output/clusters" default="/perception/object_recognition/detection/clustering/camera_lidar_fusion/clusters"/>
  <arg name="param_path" default="$(find-pkg-share autoware_image_projection_based_fusion)/config/mask_cluster_fusion.param.yaml"/>
  <arg name="sync_param_path" default="$(find-pkg-share autoware_image_projection_based_fusion)/config/fusion_common.param.yaml"/>

  <!-- for eval variable-->
  <arg name="input_rois_number" default="$(var input/rois_number)"/>

  <!-- debug -->
  <arg name="input/image0" default="/image_raw0"/>
  <arg name="input/image1" default="/image_raw1"/>
  <arg name="input/image2" default="/image_raw2"/>
  <arg name="input/image3" default="/image_raw3"/>
  <arg name="input/image4" default="/image_raw4"/>
  <arg name="input/image5" default="/image_raw5"/>
  <arg name="input/image6" default="/image_raw6"/>
  <arg name="input/image7" default="/image_raw7"/>

  <group>
    <node pkg="autoware_image_projection_based_fusion" exec="mask_cluster_fusion_node" name="mask_cluster_fusion" output="screen">
      <param name="rois_number" value="$(var input/rois_number)"/>
      <param from="$(var sync_param_path)"/>
      <param from="$(var param_path)"/>
      <remap from="input" to="$(var input/clusters)"/>
      <remap from="output" to="$(var output/clusters)"/>

      <!-- mask, camera and info -->
      <param name="input/rois0" value="$(var input/rois0)"/>
      <param name="input/camera_info0" value="$(var input/camera_info0)"/>
      <param name="input/image0" value="$(var input/image0)"/>
      <param name="input/rois1" value="$(var input/rois1)"/>
      <param name="input/camera_info1" value="$(var input/camera_info1)"/>
      <param name="input/image1" value="$(var input/image1)"/>
      <param name="input/rois2" value="$(var input/rois2)"/>
      <param name="input/camera_info2" value="$(var input/camera_info2)"/>
      <param name="input/image2" value="$(var input/image2)"/>
      <param name="input/rois3" value="$(var input/rois3)"/>
      <param name="input/camera_info3" value="$(var input/camera_info3)"/>
      <param name="input/image3" value="$(var input/image3)"/>
      <param name="input/rois4" value="$(var input/rois4)"/>
      <param name="input/camera_info4" value="$(var input/camera_info4)"/>
      <param name="input/image4" value="$(var input/image4)"/>
      <param name="input/rois5" value="$(var input/rois5)"/>
      <param name="input/camera_info5" value="$(var input/camera_info5)"/>
      <param name="input/image5" value="$(var input/image5)"/>
      <param name="input/rois6" value="$(var input/rois6)"/>
      <param name="input/camera_info6" value="$(var input/camera_info6)"/>
      <param name="input/image6" value="$(var input/image6)"/>
      <param name="input/rois7" value="$(var input/rois7)"/>
      <param name="input/camera_info7" value="$(var input/camera_info7)"/>
      <param name="input/image7" value="$(var input/image7)"/>
    </node>
  </group>
</launch>
